rstar/src/rtree.rs

```diff
@@ -327,51 +327,48 @@ where
     /// ]);
     /// let half_unit_square = AABB::from_corners([0.0, 0.0], [0.5, 1.0]);
     /// let unit_square = AABB::from_corners([0.0, 0.0], [1.0, 1.0]);
-    /// let elements_in_half_unit_square = tree.locate_in_envelope(&half_unit_square);
-    /// let elements_in_unit_square = tree.locate_in_envelope(&unit_square);
+    /// let elements_in_half_unit_square = tree.locate_in_envelope(half_unit_square);
+    /// let elements_in_unit_square = tree.locate_in_envelope(unit_square);
     /// assert_eq!(elements_in_half_unit_square.count(), 2);
     /// assert_eq!(elements_in_unit_square.count(), 3);
     /// ```
-    pub fn locate_in_envelope(&self, envelope: &T::Envelope) -> LocateInEnvelope<T> {
-        LocateInEnvelope::new(&self.root, SelectInEnvelopeFunction::new(envelope.clone()))
```

Comment:
So removing the `.clone()` call is definitely an improvement as the combination of taking a reference and cloning is inflexible. I do wonder however if we should not go the other way around and make `LocateInEnvelope` actually take a reference to the `envelope`? It already borrows from `&self` so there should not a relevant loss of functionality if it borrows from `envelope` as well?

Reply:
I don't think that's a good idea for 2 reasons:
* The lifetimes aren't the same, as the envelope is not stored as part of the tree. My use case was to have a `LocateInEnvelope` that only depended on the lifetime of the tree.
* It looks to me like the envelope is almost always a tiny struct, so from a performance perspective it should be better to just have it owned by `SelectInEnvelopeFunction` and not go through an extra reference.

It would also be a much more invasive change I think, this only changes the outer API to reflect what the internals are already doing.

Reply:
Agreed, the envelope will usually be a temporary on the stack and artificially extending its lifetime to keep the iterator viable is probably an unnecessary hassle due to the second observation.